src/lib/core/prbuf.h

```diff
@@ -87,3 +87,76 @@ prbuf_iterator_create(struct prbuf *buf, struct prbuf_iterator *iter);
  */
 int
 prbuf_iterator_next(struct prbuf_iterator *iter, struct prbuf_entry *res);
+
+struct prbuf_reader_iterator {
+	/**
+	 * File with buffer data. It is not owned by the reader.
+	 */
+	int fd;
+	/**
+	 * File offset of next record to be read in case EOF is not reached
+	 * yet.
+	 */
+	off_t pos;
+	/**
+	 * File is read ahead into `buf `and here we store the position we
+	 * read up to last time. Equals to `pos` if we don't read anything yet
+	 * or reset previous readings.
+	 */
+	off_t read_pos;
+	/**
+	 * Number of buffer bytes to be processed. It is size of unread payload
+	 * bytes. It also includes size of unused area at the end of the buffer
+	 * if we not skip it yet.
+	 *
+	 * If it is 0 then we read all the data.
+	 */
+	size_t unread_size;
+	/**
+	 * File offset of the beginning of the data area.
+	 * Data area is the area after header till the end of the buffer.
+	 */
+	off_t data_begin;
+	/**
+	 * File offset of the end of the data area.
+	 */
+	off_t data_end;
+	/*
+	 * Buffer to store data read from file.
+	 */
+	struct ibuf buf;
+};
+
+/**
+ * Initialize buffer reader.
+ * fd - file descriptor to read buffer from. It is not owned by the reader.
+ * offset - starting offset of buffer in file.
+ *
+ * Return:
+ *   0 - success
+ *  -1 - failure (diag is set)
+ */
+int
+prbuf_reader_create(struct prbuf_reader_iterator *iter, int fd, off_t offset);
+
+/**
+ * Read the next record into entry argument. If there are no more records
+ * then returned record will be a terminator (EOF, ptr == NULL && size == 0).
+ *
+ * After EOF the function can be called again and will return EOF.
+ *
+ * After failure iterator is invalid and can only be closed.
+ *
+ * Return:
+ *   0 - success
+ *  -1 - failure (diag is set)
+ */
+int
+prbuf_reader_next(struct prbuf_reader_iterator *iter,
+		  struct prbuf_entry *entry);
+
+/**
+ * Free reader resources. Should be called only once.
+ */
+void
+prbuf_reader_close(struct prbuf_reader_iterator *iter);
```

Comment:
Let's rename it to `prbuf_reader_destroy`.

Reply:
Done.